api/v1alpha1/selfnoderemediationconfig_webhook.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"os"
+	"path/filepath"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"time"
+)
+
+const (
+	webhookCertDir  = "/apiserver.local.config/certificates"
+	webhookCertName = "apiserver.crt"
+	webhookKeyName  = "apiserver.key"
+)
+
+// fields names
+const (
+	PeerApiServerTimeout = "PeerApiServerTimeout"
+	ApiServerTimeout     = "ApiServerTimeout"
+	PeerDialTimeout      = "PeerDialTimeout"
+	PeerRequestTimeout   = "PeerRequestTimeout"
+	ApiCheckInterval     = "ApiCheckInterval"
+	PeerUpdateInterval   = "PeerUpdateInterval"
+)
+
+//minimal time durations allowed string format
+const (
+	MinDurPeerApiServerTimeout = "10ms"
```

Comment:
What about using `Duration` types here? Then you can skip the `toMS` conversion and directly compare with the config value (without `MilliSeconds()`)?

Reply:
I thought about it but I also use this const in logging the error. If I used `Duration` types I would still need these consts for the err message (since in duration type it's all in milliseconds and I not as clear for the user I guess).
So I wonder is it better to have more constants and not use the toMS OR use the toMS and have less constants? @slintes

Reply:
you can format to whatever you need for the error message, e.g.
```
	test := 2000 * time.Millisecond
	errMsg := fmt.Sprintf("time in s: %vs", test.Seconds())
	fmt.Println(errMsg)
```
prints `time in s: 2s`